code/modules/mob/living/simple_animal/guardian/guardian.dm

```diff
@@ -161,16 +235,34 @@ GLOBAL_LIST_EMPTY(parasites) //all currently existing/living guardians
 		if(cooldown >= world.time)
 			stat(null, "Manifest/Recall Cooldown Remaining: [max(round((cooldown - world.time)*0.1, 0.1), 0)] seconds")
 
-/mob/living/simple_animal/hostile/guardian/Move() //Returns to summoner if they move out of range
+
+
+/mob/living/simple_animal/hostile/guardian/Moved() //Returns to summoner if they move out of range
 	. = ..()
 	snapback()
+	for(var/I in current_abilities)
+		var/datum/guardian_abilities/S = I
+		S.move_act()
+
+/mob/living/simple_animal/hostile/guardian/OpenFire(atom/A)
+	for(var/I in current_abilities)
+		var/datum/guardian_abilities/S = I
+		S.openfire_act(A)
+
+/mob/living/simple_animal/hostile/guardian/throw_impact(atom/A)
+	for(var/I in current_abilities)
+		var/datum/guardian_abilities/S = I
+		S.impact_act(A)
 
 /mob/living/simple_animal/hostile/guardian/proc/snapback()
+	for(var/I in current_abilities)
+		var/datum/guardian_abilities/S = I
+		S.snapback_act()
```

Comment:
This needs to prevent the proc from continuing if the snapback isn't expecting it.